platform/server/src/resolvers/custom-resolvers.ts

```diff
@@ -1,161 +1,246 @@
+import { GraphQLResolveInfo } from 'graphql';
 
 export default {
-    Address: {
-        __resolveType() {
-            return null;
-        }
-    },
+  Address: {
+    __resolveType() {
+      return null;
+    }
+  },
 
-    Mutation: {
-        assignVolunteers: async (parent, variables, context: GraphbackContext, info) => {
-            if (!isAuthorizedByRole(["admin"], context)) { // only admin are allowed to create volunteer actions
-                throw new UnauthorizedError()
-            }
-        
-            const now = new Date();
-            const services = context.graphback.services;
-
-            // find volunteers and recipients with ongoing actions
-            // and store their ids somewhere so that they will not be picked as candidates
-            // for automatic scheduling
-            const { items } = await services.VolunteerAction.findBy({ status: { in: ["CREATED", "ASSIGNED"] } }, context);
-                
-            const volunteerIds: any = [];
-            const recipientIds: any = [];
-            const notCompletedVolunteerActions: any = items || [];
-            for ( const action of notCompletedVolunteerActions) {
-                volunteerIds.push(action.volunteerId);
-                recipientIds.push(action.recipientId);
-            }
-            
-            // let's retrieve volunteers who are deemed free
-            const findVolunteers = services.Volunteer.findBy(buildFilter(volunteerIds), context);
-            
-            // let's retrieve recipients who do not have action
-            const findRecipients = services.Recipient.findBy(buildFilter(recipientIds), context);
-                
-            const [{items: newVolunteers}, {items: newRecipients}] = await Promise.all([findVolunteers, findRecipients]);
-            
-            let numberOfCasesCreated = 0; 
-            if (newVolunteers!.length === 0 || newRecipients!.length == 0) {
-                return services.DailyActionPlan.create({
-                    date: now,
-                    owner: "ovp-admin", // TODO retrieve this info from Keycloak context
-                    numberOfCasesCreated,
-                    numberOfVolunteersAssigned: 0,
-                    numberOfRecipients: 0
-                }, context);
-            }
+  Mutation: {
+    assignVolunteers: async ( _, __, context: GraphbackContext, info: GraphQLResolveInfo ) => {
+      const graphback = context.graphback;
+
+      if (!isAuthorizedByRole(["admin"], context)) {
+        // only admin are allowed to create volunteer actions
+        throw new UnauthorizedError();
+      }
+
+      const now = new Date();
+
+      // find volunteers and recipients with ongoing actions
+      // and store their ids somewhere so that they will not be picked as candidates
+      // for automatic scheduling
+      const { items } = await graphback.VolunteerAction.findBy(
+        { filter: { status: { in: ["CREATED", "ASSIGNED"] } } },
+        context
```

Comment:
Pass info also to allow CRUD service to extract selected fields

Reply:
See https://github.com/aerogear/OpenVolunteerPlatform/pull/153#issuecomment-690965297 for a longer explanation